ocaml/xapi/xapi_cluster.ml

```diff
@@ -47,6 +47,29 @@ let create ~__context ~pIF ~cluster_stack ~pool_auto_join ~token_timeout ~token_
       assert_pif_prerequisites (pIF,pifrec);
       let ip = ip_of_pif (pIF,pifrec) in
 
+      let assert_pool_on_same_subnet () =
+        let are_on_same_subnet =
+          try
+            let parse_ip ip_str = Scanf.sscanf ip_str "%i.%i.%i.%i" (fun a b c d -> (a, b, c, d)) in
+            let netmask = pifrec.API.pIF_netmask |> parse_ip in
+            let to_network_addr (a1, b1, c1, d1) =
+              let (a2, b2, c2, d2) = netmask in
+              (a1 land a2, b1 land b2, c1 land c2, d1 land d2)
+            in
+            let pool_host_ips =
+               Xapi_pool_helpers.get_master_slaves_list ~__context |>
+               List.map (fun x -> Db.Host.get_address ~__context ~self:x) |>
+               List.map parse_ip
+            in
+            pool_host_ips |> List.map to_network_addr |> List.for_all (fun x -> x = List.hd pool_host_ips)
```

Comment:
Why is this the right test? This compares the masked IP of each slave against the IP of the master? Maybe add a comment.

Reply:
After googling it looked like to check some IP addresses are on the same subnet, we do a bitwise and of each ip with the netmask and verify that they all give the same answer

I'll add a comment to explain this